docs/sources/setup/migrate/migrate-storage-clients/_index.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Migrate to Thanos storage clients
+menuTitle: Migrate to Thanos storage clients
+description: Migration guide for moving from legacy storage clients to Thanos storage clients.
+weight: 
+---
+# Migrate to Thanos storage clients
+
+Loki 3.4 release introduces new object storage clients based on [Thanos Object Storage Client Go module](https://github.com/thanos-io/objstore).
+
+One of the reasons for making this change is to have a consistent storage configuration across LGTM+ stack. If you are already using Grafana Mimir or Pyroscope, you can reuse the storage configuration for setting up Loki.
+
+This is an opt-in feature with Loki 3.4 release. This would be made the default way of configuring storage in future releases and the existing storage clients will be deprecated.
+
+The new storage configuration deviates from the existing format. The following sections describe the changes in detail for each provider.
+Refer to the [thanos storage reference](https://grafana.com/docs/loki/<LOKI_VERSION>/configure/#thanos_object_store_config) to view the complete list of supported storage providers and their configuration options.
+
+### Enabling the new storage clients
+1. Enable thanos storage clients by setting `use_thanos_objstore` to `true` in the `storage_config` section or by setting `-use-thanos-objstore` flag to true.
+1. When enabled, configuration under `storage_config.object_store` takes effect instead of legacy storage configs
```

Comment:
Combining these because step 2 isn't really a separate step.  Also, need a blank space before the code block.
```suggestion
1. Enable Thanos storage clients by setting `use_thanos_objstore` to `true` in the `storage_config` section or by setting the `-use-thanos-objstore` flag to true. When enabled, configuration under `storage_config.object_store` takes effect instead of legacy storage configurations.

```